src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -487,6 +487,17 @@ protected virtual string GetInsertionText(CompletionItem item, char ch)
         // This is used to decide which provider we'd collect target type completion telemetry from.
         protected virtual bool ShouldCollectTelemetryForTargetTypeCompletion => false;
 
+        protected static CompletionChange GetCompletionChangeWithParenthesis(
```

Comment:
I think the logic here is repeated a few times else where, could you consolidate them?